SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp

```diff
@@ -115,14 +145,35 @@ SpatialEventTracer::SpatialEventTracer(const FString& WorkerId)
 	}
 
 	FolderPath = EventTracePath;
-	const FString FullFileName = FString::Printf(TEXT("EventTrace_%s_%s.trace"), *WorkerId, *FDateTime::Now().ToString());
-	const FString FilePath = FPaths::Combine(FolderPath, FullFileName);
-
 	IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+	const FString FileName = TEXT("event-log");
+	const FString FileExt = TEXT(".etlog");
 	if (PlatformFile.CreateDirectoryTree(*FolderPath))
 	{
-		UE_LOG(LogSpatialEventTracer, Log, TEXT("Capturing trace to %s."), *FilePath);
-		Stream.Reset(Io_CreateFileStream(TCHAR_TO_ANSI(*FilePath), Io_OpenMode::IO_OPEN_MODE_WRITE));
+		UE_LOG(LogSpatialEventTracer, Log, TEXT("Capturing trace file%s to %s."),
+			   (Settings->bEnableEventTracingRotatingLogs) ? TEXT("s") : TEXT(""), *FolderPath);
+
+		if (Settings->bEnableEventTracingRotatingLogs)
+		{
+			const FString FullFilePathPrefix = FPaths::Combine(FolderPath, FileName);
+			const FString FullFilePathSuffix = WorkerId + FileExt;
+
+			Io_RotatingFileStreamParameters FileParamters;
+			FileParamters.filename_prefix = TCHAR_TO_ANSI(*FullFilePathPrefix);
+			FileParamters.filename_suffix = TCHAR_TO_ANSI(*FullFilePathSuffix);
+			FileParamters.max_file_size_bytes = Settings->EventTracingRotatingLogsMaxFileSizeBytes;
```

Comment:
I wonder if we need to use EventTracingRotatingLogsMaxFileCount here?

Reply:
Yeah good catch!